crates/core/tedge_mapper/src/c8y/service_monitor.rs

```diff
@@ -6,17 +6,13 @@ use serde::Serialize;
 
 #[derive(Deserialize, Serialize, Debug)]
 pub struct HealthStatus {
-    #[serde(rename = "type", default = "default_type")]
-    pub service_type: String,
+    #[serde(rename = "type")]
+    pub service_type: Option<String>,
```

Comment:
It's a question. Why `service_type` is optional? You defined the default `service_type` as `service`. What does `None` here mean?

Reply:
When the `serde` deserializes to a string if the value is not present then it has to be set to some value. So, I use Option here. When not present it will be set to None. I can't use the default value here because it has to be picked up from the configs (tedge.toml). We can write a default function, but that function has to read the `tedge.toml` file to get the default value. This is a bit trickier because needs the base directory path for the config file. Getting that at this function is trickier and needs more code changes. So, I use this as an `Option` value then later if it's `None` will be filled with the default value that is read from the configs.

Reply:
I agree with Rina: a data structure must be meaningful and not twisted just to ease some code path.

The criteria are:
* What would be the meaning of `None` for a `service_type` ? 
* Why `service_type` would be an `Option` but not the `status`?
* Why an `Option` type if `wrap is safe here because the service_type is always having some value` ?

Reply:
Refactored the code, now using `String` instead of `Option<String>` for service type.